code/weapon/weapons.cpp

```diff
@@ -1425,14 +1426,14 @@ int parse_weapon(int subtype, bool replace, const char *filename)
 		if(optional_string("+Min Damage:")){
 			stuff_float(&wip->min_damage);
 		}
-		if(wip->min_damage > wip->damage) {
+		if(wip->min_damage > wip->damage && wip->min_damage != 0.0f) {
```

Comment:
Instead of adding checks to account for when the data might not have been overridden, why not just avoid this test completely unless the optional_string was found, by moving these entire test blocks inside the optional_string tests?